test/StakeManager.t.sol

```diff
@@ -145,6 +146,27 @@ contract UnstakeTest is StakeManagerTest {
         assertEq(ERC20(stakeToken).balanceOf(testUser), 1000);
     }
 
+    function test_UnstakeShouldReturnFund_WithLockUp() public {
+        // ensure user has funds
+        deal(stakeToken, testUser, 1000);
+        StakeVault userVault = _createTestVault(testUser);
+
+        vm.startPrank(testUser);
+        ERC20(stakeToken).approve(address(userVault), 100);
+
+        uint256 lockTime = stakeManager.MIN_LOCKUP_PERIOD();
+        userVault.stake(100, lockTime);
+        assertEq(ERC20(stakeToken).balanceOf(testUser), 900);
+
+        vm.warp(block.timestamp + lockTime + 1);
```

Comment:
Might be worth adding an `unstake()` call before `warp()` to see that it's indeed reverting.
But I'm also fine leaving this out, as we have the `test_RevertWhen_FundsLocked()` test.